CONTRIBUTING.md

```diff
@@ -1,51 +1,150 @@
+   `manim/manim`).
 
-   Then you can make a Pull Request to the Manim Community Repo from your fork, through GitHub. Make sure to select `ManimCommunity/manim` instead of `3b1b/manim` as the `base repository` and your fork and branch as `head repository` - see the picture below.
+8. Finally, make sure to use the `black` autoformatter before submitting a PR.
+   You can execute it by running `black <your_file>`.  This will make changes
+   to your code, and you will need to add/commit those changes.
+
+6. To request the ManimCommunity dev team to incorporate the changes in your
+   fork into the main repository, you can make a Pull Request to the Manim
+   Community repo from your fork. Make sure to select `ManimCommunity/manim`
+   instead of `3b1b/manim` as the `base repository` and your fork and branch as
+   `head repository` - see the picture below.
 
    ![pull-requests-example-manim-community](./readme-assets/pull-requests.PNG)
 
-Also make sure to pull from upstream/master right before making a Pull Request, resolve merge conflicts locally and only then submit the Pull Request.
+
+Note: before submitting a PR please make sure that you update the docstrings of
+any methods, classes, or modules that your PR modifies, if necessary.
+
+Note: if you are contributing new features or bug fixes, it is strongly
+preferred that you include tests that accompany them in your PR.  For this
+purpose, you can create a new file under the `tests/` folder and make sure that
+your new tests pass before submitting your PR. Please check out the [Wiki Entry for Testing](https://github.com/ManimCommunity/manim/wiki/Testing) to learn more about how our tests work. 
+
+Note: if you are contributing documentation, you must also make sure that the
+command `make doctest`, executed from the `docs/` folder terminates without
+problems. Please check out the [Wiki Entry for Documentation](https://github.com/ManimCommunity/manim/wiki//Documentation-guidelines-(WIP)) to learn more about how we prefer our code documented.
+
+
+## Other guidelines
+
+1. When submitting a PR, please make special note of whether your proposed
+   changes will result in breaking changes.
+
+2. When submitting a PR, make sure that your proposed changes are as general as
+   possible, and ready to be taken advantage of by all of manim's users.  In
+   particular, leave out any machine-specific configurations, or any personal
+   information it may contain.
+
+3. If you are a maintainer, please label issues and PRs appropriately and
+   frequently.
+
+4. When opening a new issue, if there are old issues that are related, link
+   them in your new issue (even if the old ones are closed).
+
+5. If you find stale or inactive issues that seem to be irrelevant, please post
+   a comment saying 'This issue should be closed', and a community developer
+   will take a look.
+
+6. Please do as much as possible to keep issues, PRs, and development in
+   general as tidy as possible.
```

Comment:
Here it would be nice to also add a list of these nice tags that @leotrs  suggested for code reviews like nitpicking, suggestion, etc.

Reply:
Good catch. Added link